llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -747,9 +747,9 @@ void ScheduleDAGMI::finishBlock() {
   ScheduleDAGInstrs::finishBlock();
 }
 
-/// enterRegion - Called back from MachineScheduler::runOnMachineFunction after
-/// crossing a scheduling boundary. [begin, end) includes all instructions in
-/// the region, including the boundary itself and single-instruction regions
+/// enterRegion - Called back from PostMachineScheduler::runOnMachineFunction
```

Comment:
Is this true? I think both `PostMachineScheduler` and `MachineScheduler` call this.
Should it be `MachineSchedulerBase::scheduleRegions`?

Reply:
1. `MachineScheduler::runOnMachineFunction` and `PostMachineScheduler::runOnMachineFunction` all call the `MachineSchedulerBase::scheduleRegions`. 
2. Then `MachineSchedulerBase::scheduleRegions` call the `Scheduler.enterRegion` and `Scheduler.schedule`.
3. The `Scheduler` in `MachineSchedulerBase::scheduleRegions` is from `MachineScheduler::runOnMachineFunction` or `PostMachineScheduler::runOnMachineFunction.` 
4. When `Scheduler` is from the `MachineScheduler::runOnMachineFunction`, it is return from the `MachineScheduler::createMachineScheduler()`, which get a new `ScheduleDAGMILive` pointer from the `PassConfig->createMachineScheduler(this) `or `createGenericSchedLive`. Then the  `Scheduler` is `ScheduleDAGMILive` , the  `Scheduler.enterRegion` is `ScheduleDAGMILive.enterRegion`. So, we can say the `ScheduleDAGMILive::enterRegion` is called back from  `MachineScheduler::runOnMachineFunction`. The `ScheduleDAGMILive::schedule()` is same as the `ScheduleDAGMILive::enterRegion`.
5. When `Scheduler` is from the `PostMachineScheduler::runOnMachineFunction`, it is return from the `PostMachineScheduler::createPostMachineScheduler()`, which get a new `ScheduleDAGMI` pointer from the `PassConfig->createPostMachineScheduler(this) `or `createGenericSchedPostRA`. Then the  `Scheduler` is `ScheduleDAGMI` , the  `Scheduler.enterRegion` is `ScheduleDAGMI.enterRegion`. So, we can say the `ScheduleDAGMI::enterRegion` is called back from  `PostMachineScheduler::runOnMachineFunction`.The `ScheduleDAGMI::schedule()` is same as the `ScheduleDAGMI::enterRegion`.
6. I changed the two comments from `MachineScheduler::runOnMachineFunction` to `PostMachineScheduler::runOnMachineFunction` is according the "5".

Reply:
The convention for new code is "Don’t duplicate function or class name at the beginning of the comment."  https://llvm.org/docs/CodingStandards.html#commenting

Reply:
> The convention for new code is "Don’t duplicate function or class name at the beginning of the comment." https://llvm.org/docs/CodingStandards.html#commenting

I think this is a fix for existed comments. :-)

Reply:
> The convention for new code is "Don’t duplicate function or class name at the beginning of the comment." https://llvm.org/docs/CodingStandards.html#commenting

This pr just fixs three existed comments.
There are many comments in the two files need to update, if follow the   "Don’t duplicate function or class name at the beginning of the comment." Maybe need a new pr to update them.